### PR TITLE
feature dump into main

### PR DESCRIPTION
**Pull request by winglian**
- support for Galactica @ehartford 
- support specific data files from hf datasets
- drops data points from dataset if input, attention mask and label tensors don't match in length
- ignore system prompt from sharegpt dataset
- cache prepared datasets right after tokenization to allow for more reconfiguration without rebuilding cache
- support dataset shards for training
- require explicit 8bit Adam in config for optimizer

### CODE REVIEW

**Comment by ehartford** [comment]
Thank you!

**Comment by ehartford** [comment]
addresses https://github.com/winglian/axolotl/issues/7
